internal/transport/http2_client.go

```diff
@@ -978,9 +978,9 @@ func (t *http2Client) handleRSTStream(f *http2.RSTStreamFrame) {
 		statusCode = codes.Unknown
 	}
 	if statusCode == codes.Canceled {
-		// Our deadline was already exceeded, and that was likely the cause of
-		// this cancelation.  Alter the status code accordingly.
-		if d, ok := s.ctx.Deadline(); ok && d.After(time.Now()) {
+		if d, ok := s.ctx.Deadline(); ok && d.Before(time.Now()) {
```

Comment:
I think we want to include if `d == now`.

Reply:
If the current time *is* the same as the deadline, was the deadline "exceeded"?

I don't think this will matter much either way.  Done.

Reply:
deadline - now() = timeout. So any timer created is triggering when deadline == now (but yes, it may be delayed). Yes, that matters a lot less than the general fix here.